src/sys/wait.rs

```diff
@@ -46,6 +48,30 @@ libc_bitflags!(
     }
 );
 
+/// Possible child targets of`waitpid()`.
+///
+/// The most common variants for waiting are waiting
+/// for a specific child via `WaitPid::Child(pid)` and
+/// waiting for any child via `WaitPid::AnyChild`.
+#[derive(Eq, PartialEq, Clone, Copy, Debug)]
+pub enum WaitTarget {
+    /// Waiting for a child with `Pid` process id.
+    Child(Pid),
+    /// Waiting for a child of the same process group as the calling process.
+    ThisGroupChild,
+    /// Waiting for any child.
+    AnyChild,
+    /// Waiting for a child of process group `Pid` (NB: do not negate this value,
+    /// just use process id of the group leader).
+    ChildOfGroup(Pid),
+}
+
+impl From<Pid> for WaitTarget {
```

Comment:
We probably want this to follow the mapping that `waitpid` uses internally, where only positive PID values are mapped to `WaitTarget::Child`, the other 3 mappings should also be here. This should make `waitpid` work safely with current code. Additionally it should make this a backwards-compatible change.